### PR TITLE
Mes 2938 populate extra fields

### PR DESCRIPTION
**Pull request by szabikr**
Populating extra fields when test starts
Displaying those extra fields on view test result page

IMG_0133

IMG_0132

### CODE REVIEW

File: src/pages/test-results-search/components/search-result/search-result.ts [hunk 1]
```diff
@@ -33,6 +33,9 @@ export class SearchResultComponent {
   openTestResult(): void {
     // Modals are at the same level as the ion-nav so are not getting the zoom level class,
     // this needs to be passed in the create options.
+
+    console.log('### We are here in openTestResult()');
```
Comment (ghost) [inline src/pages/test-results-search/components/search-result/search-result.ts]: Debug logs.

File: src/pages/test-results-search/components/search-result/search-result.ts [hunk 2]
```diff
@@ -41,5 +44,7 @@ export class SearchResultComponent {
     );
 
     testResultModal.present();
+
+    console.log('### We are here after the testResultModal.present');
```
Comment (ghost) [inline src/pages/test-results-search/components/search-result/search-result.ts]: Debug logs.

**Review by ghost — APPROVED** [review]
Looks good.

**Comment by ghost** [comment]
Got a screenshot?

File: src/modules/tests/test-slot-attributes/test-slot-attributes.selector.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { TestSlotAttributes } from '@dvsa/mes-test-schema/categories/B';
 import { DateTime } from '../../../shared/helpers/date-time';
+import { getSlotType } from '../../../shared/helpers/get-slot-type';
```
Comment (TomBillingtonUK) [inline src/modules/tests/test-slot-attributes/test-slot-attributes.selector.ts:3]: Missing unit tests

File: src/pages/view-test-result/components/test-details-card/test-details-card.html [hunk 4]
```diff
@@ -28,14 +28,50 @@ <h4>Test details</h4>
...
+          <label>Slot type</label>
+        </ion-col>
+        <ion-col col-56>
+          <span class="mes-data">{{ data.slotType }}</span>
+        </ion-col>
+      </ion-row>
+      <ion-row class="mes-data-row-with-separator">
+        <ion-col col-40>
+          <label>Special requirements</label>
+        </ion-col>
```
Comment (TomBillingtonUK) [inline src/pages/view-test-result/components/test-details-card/test-details-card.html:56]: You seem to be missing the indicators

File: src/pages/view-test-result/view-test-result.ts [hunk 5]
```diff
@@ -85,7 +85,6 @@ export class ViewTestResultPage extends BasePageComponent implements OnInit {
     super(platform, navController, authenticationProvider);
```
Comment (TomBillingtonUK) [inline src/pages/view-test-result/view-test-result.ts:85]: No unit tests for changes

File: src/shared/helpers/get-slot-type.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { TestSlot } from '@dvsa/mes-journal-schema';
```
Comment (TomBillingtonUK) [inline src/shared/helpers/get-slot-type.ts:1]: No unit tests for this file

File: src/pages/test-results-search/components/search-result/search-result.ts (hunk 1 above)
Comment (szabikr) [reply src/pages/test-results-search/components/search-result/search-result.ts]: removed them

File: src/pages/test-results-search/components/search-result/search-result.ts (hunk 2 above)
Comment (szabikr) [reply src/pages/test-results-search/components/search-result/search-result.ts]: removed them

File: src/modules/tests/test-slot-attributes/test-slot-attributes.selector.ts (hunk 3 above)
Comment (szabikr) [reply src/modules/tests/test-slot-attributes/test-slot-attributes.selector.ts:3]: wrote the unit tests

File: src/pages/view-test-result/components/test-details-card/test-details-card.html (hunk 4 above)
Comment (szabikr) [reply src/pages/view-test-result/components/test-details-card/test-details-card.html:56]: place the indicators where needed

File: src/shared/helpers/get-slot-type.ts (hunk 6 above)
Comment (szabikr) [reply src/shared/helpers/get-slot-type.ts:1]: wrote the unit tests

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**